File: 백준/Gold/11509. 풍선 맞추기/풍선 맞추기.py
import sys

# initialization
n: int = int(sys.stdin.readline())
nums: list[int] = list(map(int, sys.stdin.readline().split(" ")))
arrows: list[int] = []

# for all balloons, if there's no arrow of balloon's height then append with h-1, 
# else sub 1 to that arrow's height
for x in nums:
    hit: bool = False
    for i in range(len(arrows)):
        if x == arrows[i]:
            arrows[i] -= 1
            hit = True
            break
    if not arrows or not hit:
        arrows.append(x-1)

sys.stdout.write(str(len(arrows)))

File: 백준/Gold/11509. 풍선 맞추기/풍선 맞추기.rs
use std::io;

fn main() {
    let mut buf: String = String::new();
    io::stdin().read_line(&mut buf).unwrap();
    let n: usize = buf.trim().parse().unwrap();

    let mut buf: String = String::new();
    io::stdin().read_line(&mut buf).unwrap();
    let nums: Vec<isize> = buf.trim().split(' ')
        .map(|c| c.parse::<isize>().unwrap()).collect();
    let mut arrows: Vec<isize> = Vec::with_capacity(n);

    for x in nums {
        let mut hit: bool = false;
        for i in 0..arrows.len() {
            if *arrows.get(i).unwrap() == x {
                *arrows.get_mut(i).unwrap() -= 1;
                hit = true;
                break;
            }
        }
        if arrows.is_empty() || !hit {
            arrows.push(x-1);
        }
    }

    println!("{}", arrows.len());
}